src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -199,6 +199,45 @@ public override void VisitOperatorDeclaration(OperatorDeclarationSyntax node)
             Visit(node.ExpressionBody);
         }
 
+        public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+        {
+            if (node.MayBeNameofOperator())
+            {
+                WithTypeParametersBinder nextWhenNameofOperator = ((_enclosing.Flags & BinderFlags.InContextualAttributeBinder) != 0) && MessageID.IDS_FeatureExtendedNameofScope.IsFeatureAvailable(node)
+                    ? getExtraWithTypeParametersBinder(_enclosing, getAttributeTarget(_enclosing))
+                    : null;
+
+                var argumentExpression = node.ArgumentList.Arguments[0].Expression;
+                var possibleNameofBinder = new NameofBinder(argumentExpression, _enclosing, nextWhenNameofOperator);
+
+                AddToMap(node, possibleNameofBinder);
+            }
+
+            base.VisitInvocationExpression(node);
```

Comment:
>base.VisitInvocationExpression(node); [](http://example.com/codeflow?start=12&length=37)

If we created NameofBinder above, I think we need to change/restore ```_enclosing``` around this call. Whatever binders we might be creating during this visit, they should be linked to the NameofBinder.

Reply:
I'd totally missed that. Thanks